packages/context/src/__tests__/unit/context-view.unit.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/context
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {Binding, BindingScope, filterByTag, Context, ContextView} from '../..';
+
+describe('ContextView', () => {
+  let app: Context;
+  let server: Context;
+
+  let bindings: Binding<unknown>[];
+  let contextView: ContextView;
+
+  beforeEach(givenContextView);
+
+  it('tracks bindings', () => {
+    expect(contextView.bindings).to.eql(bindings);
+  });
+
+  it('resolves bindings', async () => {
+    expect(await contextView.resolve()).to.eql(['BAR', 'FOO']);
+    expect(await contextView.values()).to.eql(['BAR', 'FOO']);
+  });
+
+  it('resolves bindings as a getter', async () => {
+    expect(await contextView.asGetter()()).to.eql(['BAR', 'FOO']);
+  });
+
+  it('reloads bindings after reset', async () => {
+    contextView.reset();
+    const abcBinding = server
+      .bind('abc')
+      .to('ABC')
+      .tag('abc');
+    const xyzBinding = server
+      .bind('xyz')
+      .to('XYZ')
+      .tag('foo');
+    expect(contextView.bindings).to.containEql(xyzBinding);
+    // `abc` does not have the matching tag
+    expect(contextView.bindings).to.not.containEql(abcBinding);
+    expect(await contextView.values()).to.eql(['BAR', 'XYZ', 'FOO']);
```

Comment:
Hmm, doesn't `reset` mean `BAR` and `FOO` should be removed?

Reply:
reset() cleans up the cache to force `contextView.values()` to pick up latest bindings.

Reply:
I renamed `reset` to be `refresh` to avoid confusions.

Reply:
@raymondfeng Great `refresh` sounds more clear to me. And no confusions any more when I read the implementation code.